TestApp/Search/Sources/SearchViews/SearchFlow.swift

```diff
@@ -0,0 +1,51 @@
+import SwiftUI
+import Navigation
+import Search
+
+public struct SearchFlow: View {
+
+    enum Path: Hashable {
+        case results
+    }
+
+    @StateObject private var path = PushNavigationPath()
+    @Binding private var popToRoot: Bool
+
+    let booking: (_ token: String) -> Void
+
+    public var body: some View {
+        PushNavigationStack(path: path) {
+            search
+        }
+        .popsToRoot(path, trigger: $popToRoot)
+        .destination(for: Path.self) { path in
```

Comment:
If possible, the API consistency for "pushed destinations" could be unified:

1. As with other modifiers overriding the native version, it could be named the same as `navigationDestination`
2. The `navigationDestination` could use the same API as `sheet` etc - ie., where instead of having to switch the case inside the closure, we could just call `navigationDestination(for Path.self, value: .results) { results }` for each value we want to assign the destination. Is the reason associated values - but if yes, how do they work with `sheet` etc?
3. The naming now varies between `destination` vs. `path` vs. `route`. We could specify that a Destination is the "next destination" from the screen, while path (or route) is a full "path", specifying an explicit instance of sequence of destinations.

Reply:
1. it could, though we aren't actually limited to push navigation with these - we can use whatever is "registered" like this anywhere, even in sheets and the like. We don't have to, but it is possible. But maybe we should just not encourage that?
2. I think I had that at one point. The way it is here, we have a compile-time guarantee that all cases are handled, which we would lose if we split it. I think it's a bit different for modals because there, different cases might be using different modals - sheets vs alerts for example.
3. Agreed, there was a lot of iteration in this area, we should unify it.

Reply:
1) We should limit it to push navigation. Not only to better match the native api, but to have consistent expectations compared to explicit sheet and other modifiers.

For 2), we may want to decide between one way or the other. Now the added benefit of compile safety of destinations is only partial, when sheets do not follow it as well.

Personally, the API that sheets now have feel more flexible to me (to avoid switching in the closure), even if we would lose the compilation safety for push screens.

Reply:
Alright, so to sum up some slack discussions for the benefit of everyone else:

There are two different methods for using the iOS 16 navigation.

1) [navigationDestination(for:destination:)](https://developer.apple.com/documentation/swiftui/view/navigationdestination(for:destination:)) - this takes a type and is triggered by updating the path in the stack. We have this working in this project and it also allows us to push/pop multiple screens "at once" (+/- a delay that needs to wait for the intermediate animations).

2) [navigationDestination(isPresented:destination:)](https://developer.apple.com/documentation/swiftui/view/navigationdestination(ispresented:destination:)) - this takes a binding, which is used for triggering the presentation/dismissal. It is limited to showing a single screen, because it's a simple true/false flag. We'd need to have a separate binding for each such presentation, which doesn't really work with the stack. There would be no way to push more of the same type of screen. There would be no way to really ensure the ordering of those screens if you flipped multiple `Bool`s at once.

These two methods don't seem to work with each other. You can try this for yourself - the 2. mentioned above when presented **doesn't alter the stack's path** - it's completely separate from that:
```swift
import SwiftUI

struct ContentView: View {

    @State var navigationPath = NavigationPath()
    @State var isPresented = false

    var body: some View {
        NavigationStack(path: $navigationPath) {
            VStack(spacing: 20) {
                Button("Go to Detail") {
                    isPresented = true
                }

                Button("Go to 3") {
                    navigationPath.append(3)
                }
            }
            .navigationDestination(isPresented: $isPresented) {
                Text("detail")
            }
            .navigationDestination(for: Int.self) { int in
                Text(String(int))
            }
        }
        .onChange(of: navigationPath.count) { numberOfItems in
            print("numberOfItems: \(numberOfItems)")
        }
    }
}

struct ContentView_Previews: PreviewProvider {
    static var previews: some View {
        ContentView()
    }
}
```

So it seems like there is in fact no way to use 2. with a stack. And we do want the stack - otherwise, we wouldn't even need to hack together a version of the iOS 16 API.